### PR TITLE
[BEAM-1935] Improve Work Rejection handling

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
The timing between checking for a shutdown state and submitting work is
racy. Re-check to see if a work rejection is acceptable because the
underlying executor is shut down before throwing a
RejectedExecutionException.

Renames to internal implementation classes of TransformExecutorServices.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @ssisk

**Comment by tgroh** [comment]
retest this please

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutorServices.java [hunk 1]
```diff
@@ -55,18 +58,36 @@ public static TransformExecutorService serial(ExecutorService executor) {
...
     }
 
     @Override
     public void schedule(TransformExecutor<?> work) {
       if (active.get()) {
-        executor.submit(work);
+        try {
+          executor.submit(work);
+        } catch (RejectedExecutionException rejected) {
+          boolean stillActive = active.get();
```
Comment (ssisk) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutorServices.java]: is it ever possible for active to go: true -> false -> true, and thus cause this to also be racy?

Comment (tgroh) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutorServices.java]: It is not possible with the API of `TransformExecutorService` to ever un-shutdown an executor

**Pull request closed without merging**